### ISSUE TITLE
Layout/LineContinuationSpacing applies to non-Ruby source (DATA section)

### ISSUE THREAD

**Issue by Bo98** [rubocop/rubocop#10763]
The new `Layout/LineContinuationSpacing` cop incorrectly applies itself to text after `__END__`.

--------

## Expected behavior

The cop is not raised for non-Ruby code.

## Actual behavior

The cop is triggered.

## Steps to reproduce the problem

```rb
# frozen_string_literal: true

class ThisIsRubyCode
  def initialize
    puts "Hello"
  end
end

__END__
random text, such as a git diff           \
which should not be checked
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.31.0 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 2.6.8 x86_64-darwin15)
  - rubocop-performance 1.14.2
  - rubocop-rails 2.15.0
```
